solidity/test/LPRewardsTest.js

```diff
@@ -0,0 +1,219 @@
+      let wrappedTokenStakerBalance2 = await wrappedToken.balanceOf(staker2)
+      expect(wrappedTokenStakerBalance2).to.eq.BN(initialWrappedTokenStakerBallance2)
+
+      await lpRewards.stake(web3.utils.toBN(4000).mul(tokenDecimalMultiplier), {
+        from: staker1,
+      })
+      await lpRewards.stake(web3.utils.toBN(5000).mul(tokenDecimalMultiplier), {
+        from: staker2,
+      })
+
+      wrappedTokenBalance = await wrappedToken.balanceOf(lpRewards.address)
+      // 4,000 + 5,000 = 9,000
+      expect(wrappedTokenBalance).to.eq.BN(
+        web3.utils.toBN(9000).mul(tokenDecimalMultiplier)
+      )
+
+      wrappedTokenStakerBalance1 = await wrappedToken.balanceOf(staker1)
+      // 10,000 - 4,000 = 6,000
+      expect(wrappedTokenStakerBalance1).to.eq.BN(
+        web3.utils.toBN(6000).mul(tokenDecimalMultiplier)
+      )
+
+      wrappedTokenStakerBalance2 = await wrappedToken.balanceOf(staker2)
+      // 20,000 - 5,000 = 15,000
+      expect(wrappedTokenStakerBalance2).to.eq.BN(
+        web3.utils.toBN(15000).mul(tokenDecimalMultiplier)
+      )
+    })
+  })
+
+  describe("tokens distribution", () => {
+    const precision = 1
+
+    it("should be possible to check earned rewards", async () => {
+      const keepRewards = new BN(10000).mul(tokenDecimalMultiplier)
+      const wrappedTokenStakerBallance = web3.utils
+        .toBN(10000)
+        .mul(tokenDecimalMultiplier)
+
+      await fundKEEPReward(lpRewards.address, keepRewards)
+      await mintAndApproveWrappedTokens(
+        wrappedToken,
+        lpRewards.address,
+        staker1,
+        wrappedTokenStakerBallance
+      )
+
+      await lpRewards.setRewardDistribution(rewardDistribution)
+      await lpRewards.notifyRewardAmount(keepRewards, {from: rewardDistribution})
+
+      const wrappedTokensToStake = web3.utils
+        .toBN(2000)
+        .mul(tokenDecimalMultiplier)
+      await lpRewards.stake(wrappedTokensToStake, {from: staker1})
+
+      const future = (await time.latest()).add(time.duration.days(7))
+      await timeIncreaseTo(future)
+
+      const actualEarnings = (await lpRewards.earned(staker1)).div(
+        tokenDecimalMultiplier
+      )
+      const expectedEarnings = new BN(10000)
+      expect(actualEarnings).to.gte.BN(expectedEarnings.subn(precision))
+      expect(actualEarnings).to.lte.BN(expectedEarnings.addn(precision))
+
+      // 10,000 / 2,000 = 5 KEEP,
+      // Contract function will return 4999999999999999881 KEEP. Solidity
+      // does not have floating numbers and when dividing by 10^18, precision will
+      // not be on point.
+      const actualRewardPerToken = (await lpRewards.rewardPerToken()).div(
+        tokenDecimalMultiplier
+      )
+      const expectedRewardPerWrappedToken = new BN(5)
+      expect(actualRewardPerToken).to.gte.BN(
+        expectedRewardPerWrappedToken.subn(precision)
+      )
+      expect(actualRewardPerToken).to.lte.BN(
+        expectedRewardPerWrappedToken.addn(precision)
+      )
```

Comment:
Woud could try to calculate the exact amount like in the example below. But also we can hardcode the expected value.

```js
      const duration = await lpRewards.DURATION.call()
      const rewardRate = keepRewards.div(duration)

      const expectedRewardPerWrappedToken = rewardRate
        .mul(duration)
        .mul(tokenDecimalMultiplier)
        .div(stakedAmount)
```

Reply:
Hardcoding can be the cleaniest solution. Of course we need to add a comment next to the value definition with steps how it was calculated.

Reply:
I've added a comment showing the math behind a reward per token. b0165bd

I think checking math on contract with a simple expected value in test should work, especially this is considered to be a trusted source.